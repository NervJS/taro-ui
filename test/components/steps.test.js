import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import AtSteps from '../../.temp/components/steps/index'

describe('AtSteps Snap', () => {
  const items1 = [
    {
      title: '步骤一',
      desc: '这里是额外的信息，最多两行',
      icon: {
        value: 'sound',
        activeColor: '#fff',
        inactiveColor: '#78A4FA',
        size: '14'
      }
    },
    {
      title: '步骤二',
      desc: '这里是额外的信息，最多两行',
      icon: {
        value: 'shopping-cart',
        activeColor: '#fff',
        inactiveColor: '#78A4FA',
        size: '14'
      }
    },
    {
      title: '步骤三',
      desc: '这里是额外的信息，最多两行',
      icon: {
        value: 'camera',
        activeColor: '#fff',
        inactiveColor: '#78A4FA',
        size: '14'
      }
    }
  ]

  const items2 = [
    {
      title: '步骤一',
      desc: '这里是额外的信息，最多两行',
      success: true
    },
    {
      title: '步骤二',
      desc: '这里是额外的信息，最多两行'
    },
    {
      title: '步骤三',
      desc: '这里是额外的信息，最多两行',
      error: true
    }
  ]

  it('render initial AtSteps', () => {
    const componet = renderToString(<AtSteps />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSteps -- props customStyle', () => {
    const componet = renderToString(<AtSteps customStyle='color:red;' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSteps -- props className', () => {
    const componet = renderToString(<AtSteps className='test' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSteps -- props items', () => {
    const componet1 = renderToString(<AtSteps items={items1} />)
    expect(componet1).toMatchSnapshot()
    const componet2 = renderToString(<AtSteps items={items2} />)
    expect(componet2).toMatchSnapshot()
  })

  it('render AtSteps -- props current', () => {
    const componet1 = renderToString(<AtSteps items={items1} current={1} />)
    expect(componet1).toMatchSnapshot()
  })
})
