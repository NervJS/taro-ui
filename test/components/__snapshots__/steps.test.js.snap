// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtSteps Snap render AtSteps -- props className 1`] = `"<div class=\\"at-steps test\\"></div>"`;

exports[`AtSteps Snap render AtSteps -- props current 1`] = `"<div class=\\"at-steps\\"><div class=\\"at-steps__item at-steps__item--inactive\\"><div class=\\"at-steps__circular-wrap\\">undefined<div class=\\"at-steps__circular\\"><span class=\\"taro-text at-icon at-icon-sound at-steps__circle-icon\\"></span></div><div class=\\"at-steps__right-line\\"></div></div><div class=\\"at-steps__title\\">步骤一</div><div class=\\"at-steps__desc\\">这里是额外的信息，最多两行</div></div><div class=\\"at-steps__item at-steps__item--active\\"><div class=\\"at-steps__circular-wrap\\"><div class=\\"at-steps__left-line\\"></div><div class=\\"at-steps__circular\\"><span class=\\"taro-text at-icon at-icon-shopping-cart at-steps__circle-icon\\"></span></div><div class=\\"at-steps__right-line\\"></div></div><div class=\\"at-steps__title\\">步骤二</div><div class=\\"at-steps__desc\\">这里是额外的信息，最多两行</div></div><div class=\\"at-steps__item at-steps__item--inactive\\"><div class=\\"at-steps__circular-wrap\\"><div class=\\"at-steps__left-line\\"></div><div class=\\"at-steps__circular\\"><span class=\\"taro-text at-icon at-icon-camera at-steps__circle-icon\\"></span></div>undefined</div><div class=\\"at-steps__title\\">步骤三</div><div class=\\"at-steps__desc\\">这里是额外的信息，最多两行</div></div></div>"`;

exports[`AtSteps Snap render AtSteps -- props customStyle 1`] = `"<div style=\\"color:red;\\" class=\\"at-steps\\"></div>"`;

exports[`AtSteps Snap render AtSteps -- props items 1`] = `"<div class=\\"at-steps\\"><div class=\\"at-steps__item at-steps__item--active\\"><div class=\\"at-steps__circular-wrap\\">undefined<div class=\\"at-steps__circular\\"><span class=\\"taro-text at-icon at-icon-sound at-steps__circle-icon\\"></span></div><div class=\\"at-steps__right-line\\"></div></div><div class=\\"at-steps__title\\">步骤一</div><div class=\\"at-steps__desc\\">这里是额外的信息，最多两行</div></div><div class=\\"at-steps__item at-steps__item--inactive\\"><div class=\\"at-steps__circular-wrap\\"><div class=\\"at-steps__left-line\\"></div><div class=\\"at-steps__circular\\"><span class=\\"taro-text at-icon at-icon-shopping-cart at-steps__circle-icon\\"></span></div><div class=\\"at-steps__right-line\\"></div></div><div class=\\"at-steps__title\\">步骤二</div><div class=\\"at-steps__desc\\">这里是额外的信息，最多两行</div></div><div class=\\"at-steps__item at-steps__item--inactive\\"><div class=\\"at-steps__circular-wrap\\"><div class=\\"at-steps__left-line\\"></div><div class=\\"at-steps__circular\\"><span class=\\"taro-text at-icon at-icon-camera at-steps__circle-icon\\"></span></div>undefined</div><div class=\\"at-steps__title\\">步骤三</div><div class=\\"at-steps__desc\\">这里是额外的信息，最多两行</div></div></div>"`;

exports[`AtSteps Snap render AtSteps -- props items 2`] = `"<div class=\\"at-steps\\"><div class=\\"at-steps__item at-steps__item--active\\"><div class=\\"at-steps__circular-wrap\\">undefined<div class=\\"at-steps__circular\\"><span class=\\"taro-text at-steps__num\\">1</span></div><div class=\\"at-steps__right-line\\"></div></div><div class=\\"at-steps__title\\">步骤一</div><div class=\\"at-steps__desc\\">这里是额外的信息，最多两行</div></div><div class=\\"at-steps__item at-steps__item--inactive\\"><div class=\\"at-steps__circular-wrap\\"><div class=\\"at-steps__left-line\\"></div><div class=\\"at-steps__circular\\"><span class=\\"taro-text at-steps__num\\">2</span></div><div class=\\"at-steps__right-line\\"></div></div><div class=\\"at-steps__title\\">步骤二</div><div class=\\"at-steps__desc\\">这里是额外的信息，最多两行</div></div><div class=\\"at-steps__item at-steps__item--inactive\\"><div class=\\"at-steps__circular-wrap\\"><div class=\\"at-steps__left-line\\"></div><div class=\\"at-steps__circular\\"><span class=\\"taro-text at-steps__num\\">3</span></div>undefined</div><div class=\\"at-steps__title\\">步骤三</div><div class=\\"at-steps__desc\\">这里是额外的信息，最多两行</div></div></div>"`;

exports[`AtSteps Snap render initial AtSteps 1`] = `"<div class=\\"at-steps\\"></div>"`;
